scripts/chk_shellscripts/chk_shellscripts.sh

```diff
@@ -17,8 +17,8 @@ find . -type f -name "*.sh" | sort >"${FOUND_FILES_TMP}"
 
 SHELLCHECK=${SHELLCHECK:-"$(command -v -- shellcheck)"}
 if [ ! -f "${SHELLCHECK}" ]; then
-  echo "error: shellcheck '${SHELLCHECK}' not found."
-  exit 1
+    echo "error: shellcheck '${SHELLCHECK}' not found."
```

Comment:
Why single quote?

Reply:
Probably to distinguish it from the rest of the text. They're not interpreted by shell there. Once inside `"` Bash only interprets `$var` as variables, single quotes are just normal text without any syntactical meaning.

Reply:
Just curious: When exactly is a double quoting of bash variable names warranted?

Reply:
Almost always. It's easier to ask when it's not warranted :) Bash variables work almost like preprocessor macros in C++ - if you don't quote them, spaces inside will get interpreted the same way as spaces that separate arguments. So something like `$x` that looks like a single argument can end up being multiple arguments if it contains spaces. It's not exactly the same as preprocessor though because quotes inside variables are treated specially. If you have e.g. `x='bc" "de'` and then do `echo "a${x}f"`, it will still end up as a single argument (`abc" "def`) rather than two (`abc` `def`).

But note that the `$SHELLCHECK` variable above **is** double quoted - along with all the surrounding text. If it contains spaces, it does not really matter because the whole `error: shellcheck '${SHELLCHECK}' not found.` will end up being a single argument anyway. So you could say, you don't need to quote the variable itself separately if it's already inside a quoted string. And like I said above the single quotes do not count as quoting here - they'll just text and will be included in the output.

One important quirk is that if the string contains nested constructs like `$()`, you do need to quote stuff inside them. So for example in `echo "error: $(cat "$file")"` the `$file` variable needs to be quoted separately because it's not really a part of the outer string. There are two separate commands - `cat "$file"` gets executed first and only its output is inserted into the outer string that then goes to `echo`.